backend/grant/admin/views.py

```diff
@@ -712,3 +712,103 @@ def edit_comment(comment_id, hidden, reported):
 
     db.session.commit()
     return admin_comment_schema.dump(comment)
+
+
+# Financials
+
+@blueprint.route("/financials", methods=["GET"])
+@admin.admin_auth_required
+def financials():
+    grants = {
+        'total': '0',
+        'matching': '0',
+        'bounty': '0',
+    }
+    contributions = {
+        'total': '0',
+        'staking': '0',
+        'funding': '0',
+        'funded': '0',
+        'refunding': '0',
+        'refunded': '0',
+        'gross': '0',
+        'by_stage': {},
+    }
+    payouts = {
+        'total': '0',
+        'due': '0',
+        'paid': '0',
+        'future': '0',
+    }
+
+    def add_str_dec(a: str, b: str):
+        return str(Decimal(a) + Decimal(b))
+
+    def add_cont(amount, stage, is_refunded, is_staking):
+        contributions['total'] = add_str_dec(contributions['total'], amount)
+        if is_staking:
+            contributions['staking'] = add_str_dec(contributions['staking'], amount)
+            # staking only counts towards total & staking
+            return
+
+        if stage in contributions['by_stage']:
+            contributions['by_stage'][stage] = add_str_dec(contributions['by_stage'][stage], amount)
+        else:
+            contributions['by_stage'][stage] = add_str_dec('0', amount)
+
+        if stage in [ProposalStage.FUNDING_REQUIRED]:
+            contributions['funding'] = add_str_dec(contributions['funding'], amount)
+        if stage in [ProposalStage.COMPLETED, ProposalStage.WIP]:
+            contributions['funded'] = add_str_dec(contributions['funded'], amount)
+        if stage in [ProposalStage.CANCELED, ProposalStage.FAILED]:
+            if is_refunded:
+                contributions['refunded'] = add_str_dec(contributions['refunded'], amount)
+            else:
+                contributions['refunding'] = add_str_dec(contributions['refunding'], amount)
+
+    proposals = Proposal.query.all()
+
+    for p in proposals:
+        if p.status == ProposalStatus.LIVE:
+            if p.stage in [ProposalStage.WIP, ProposalStage.COMPLETED]:
```

Comment:
Deleted proposals and contributions for staking proposals that get rejected still need to be accounted for, I think we should be querying for contributions rather than proposals.

Reply:
More readable to, I think. And less work/data exchange to python.